encoding/ccf/decode.go

```diff
@@ -2182,9 +2182,15 @@ func (d *Decoder) decodeTypeParameterTypeValue(visited *cadenceTypeByCCFTypeID)
 		return cadence.TypeParameter{}, err
 	}
 
+	var typeBound cadence.TypeBound
+	if t != nil {
+		typeBound = cadence.NewSubtypeTypeBound(t)
+	}
+
 	// Unmetered because decodeTypeParamTypeValue is metered in decodeTypeParamTypeValues and called nowhere else
 	// Type is metered.
-	return cadence.NewTypeParameter(name, t), nil
+	// TODO: implement this for generalized bounds
+	return cadence.NewTypeParameter(name, typeBound), nil
```

Comment:
This should be sufficient to support decoding any function types that aren't one of the small set of builtins that this PR changes.